components/script/dom/gpubuffer.rs

```diff
@@ -147,15 +149,19 @@ impl GPUBufferMethods for GPUBuffer {
             // Step 3
             GPUBufferState::Mapped | GPUBufferState::MappedAtCreation => {
                 let mut info = self.map_info.borrow_mut();
-                let m_info = info.as_mut().unwrap();
+                let m_info = if let Some(m_info) = info.as_mut() {
+                    m_info
+                } else {
+                    return Err(Error::Operation);
+                };
                 let m_range = m_info.mapping_range.clone();
                 if let Err(e) = self.channel.0.send((
                     self.device.use_current_scope(),
                     WebGPURequest::UnmapBuffer {
                         buffer_id: self.id().0,
                         device_id: self.device.id().0,
                         array_buffer: IpcSharedMemory::from_bytes(
-                            m_info.mapping.borrow().as_slice(),
+                            m_info.mapping.lock().unwrap().borrow().as_slice(),
```

Comment:
Tell me if I'm wrong but I believe you can just do `&m_info.mapping.lock().unwrap()`, thanks to [deref](https://doc.rust-lang.org/std/ops/trait.Deref.html).

Reply:
I think you meant:

`m_info.mapping.lock().unwrap().borrow()`

Reply:
yes you're right I think, although you could I think also remove the `borrow()` if you were to also deref the guard, like `&*m_info.mapping.lock().unwrap()`. Anyway this is not very important.